test/e2e/app-dir/app-inline-css/index.test.ts

```diff
@@ -15,5 +14,37 @@ describe('app dir - css - experimental inline css', () => {
       const p = await browser.elementByCss('p')
       expect(await p.getComputedCss('color')).toBe('rgb(255, 255, 0)') // yellow
     })
+
+    it('should not return rsc payload with inlined style as a dynamic client nav', async () => {
+      let rscRequest
+
+      const browser = await next.browser('/', {
+        beforePageLoad: async (page) => {
+          page.on('request', (request) => {
+            if (request.url().includes('_rsc')) {
+              rscRequest = request
+            }
+          })
+        },
+      })
+
+      await browser.waitForElementByCss('#link-a').click()
+      await browser.waitForElementByCss('#page-a')
+
+      expect(await rscRequest.response()).not.toContain('color')
```

Comment:
Do we need a retry here just in case there's a race between `page.on('request')` and `waitForElement` resolving?